module/zfs/dsl_scan.c

```diff
@@ -1104,7 +1212,7 @@ dsl_scan_zil_block(zilog_t *zilog, blkptr_t *bp, void *arg, uint64_t claim_txg)
 	SET_BOOKMARK(&zb, zh->zh_log.blk_cksum.zc_word[ZIL_ZC_OBJSET],
 	    ZB_ZIL_OBJECT, ZB_ZIL_LEVEL, bp->blk_cksum.zc_word[ZIL_ZC_SEQ]);
 
-	VERIFY(0 == scan_funcs[scn->scn_phys.scn_func](dp, bp, &zb));
+	scan_funcs[scn->scn_phys.scn_func](dp, bp, &zb, B_FALSE);
```

Comment:
there is currently only one `scan_func_t`: `dsl_scan_scrub_cb()` which has always (as far as I can tell) returned zero. The only info I could find about its purpose is a comment indicating that returning zero means `/* do not relocate this block */` which doesn't seem to make sense anymore. Many of the callers didn't check the return code anyway, and the only ones that did simply did a `VERIFY(0 ==...)` as you see here.

As part of the cleanup I mentioned, I changed the function to return void.